code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -5,3 +5,261 @@
 
 /datum/action/xeno_action/return_to_core/action_activate()
 	SEND_SIGNAL(owner, COMSIG_XENOMORPH_CORE_RETURN)
+
+// ***************************************
+// *********** Reposition Core
+// ***************************************
+/datum/action/xeno_action/activable/reposition_core
+	name = "Reposition Core"
+	action_icon_state = "emit_neurogas"
+	mechanics_text = "Reposition your core to a target location on weeds. The further away the location is the longer this will take up to a maximum of 60 seconds. 3 minute cooldown."
+	ability_name = "reposition core"
+	plasma_cost = 150
+	cooldown_timer = 600 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybind_signal = COMSIG_XENOABILITY_REPOSITION_CORE
+
+/datum/action/xeno_action/activable/reposition_core/on_cooldown_finish()
+	owner.playsound_local(owner.loc, 'sound/voice/alien_drool1.ogg', 50, TRUE)
+	to_chat(owner, "<span class='xenonotice'>We regain the strength to reposition our neural core.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/reposition_core/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(QDELETED(target))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+	var/turf/T = get_turf(target)
+
+	if(!istype(T))
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>We can't transfer our core here!</span>")
+		return FALSE
+
+	if(!X.Adjacent(T))
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>We can only reposition to a space adjacent to us!</span>")
+		return FALSE
+
+	if(!check_build_location(T, X))
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>We can't transfer our core here!</span>")
+		return FALSE
```

Comment:
Something to think on.

the `check_build_location()` has a bunch of messages to the player that will always be shown. 

thus, in `check_build_location()` you can have it return a `message` instead of `FALSE` (assuming that no message means its ok to build) and play that message in the `to_chat()` which will get processed by the `silent` check.